File: yanhuo-xo/src/main/java/com/yanhuo/xo/entity/Note.java
package com.yanhuo.xo.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.yanhuo.common.entity.BaseEntity;
import lombok.Data;

/**
 * @author xiaozhao
 */
@Data
@TableName("t_note")
public class Note extends BaseEntity {

    private String title;

    private String content;

    private String noteCover;

    private Integer noteCoverHeight;

    private String uid;

    private String cid;

    private String cpid;

    private String urls;

    //图片数量
    private Integer count;

    private Integer sort;

    //是否置顶
    private Integer pinned;

    private Integer status;

    //类型（图片或视频）
    private Integer type;

    private Long likeCount;

    private Long collectionCount;

    private Long commentCount;

    private Long viewCount;

}

File: yanhuo-web/src/type/note.ts
export interface NoteSearch {
  id: string;
  title: string;
  src: string;
  uid: string;
  username: string;
  avatar: string;
  cid: string;
  cpid: string;
  urls: string;
  count: number;
  type: number;
  likeCount: number;
  time: number | string;
}

export interface NoteInfo {
  id: string;
  title: string;
  content: string;
  noteCover: string;
  uid: string;
  username: string;
  avatar: string;
  imgList: Array<string>;
  type: number;
  likeCount: number;
  collectionCount: number;
  commentCount: number;
  tagList: Array<any>;
  time: string;
  isFollow: boolean,
  isLike: boolean,
  isCollection: boolean,
  pinned: number
}

export interface NoteDTO {
  keyword: string;
  type: number;
  cid: string;
  cpid: string;
}
